doc/contributing/coding-conventions.chapter.md

```diff
@@ -480,15 +480,10 @@ Preferred source hash type is sha256. There are several ways to get it.
 
 4. Extracting hash from local source tarball can be done with `sha256sum`. Use `nix-prefetch-url file:///path/to/tarball` if you want base32 hash.
 
-5. Fake hash: set fake hash in package expression, perform build and extract correct hash from error Nix prints.
-
-    For package updates it is enough to change one symbol to make hash fake. For new packages, you can use `lib.fakeSha256`, `lib.fakeSha512` or any other fake hash.
+5. Fake hash: set the hash to `""`, `lib.fakeHash`, `lib.fakeSha256`, or `lib.fakeSha512` in the package expression, perform build and extract correct hash from error Nix prints.  You must use one of these four fake hashes and not some randomly-chosen hash; see below for the reason.
```

Comment:
Please do not refer to "below" or "above" in web documents. The document may change structure, and the verbal reference will be left dangling if change authors don't adapt it manually. People make mistakes, let's avoid opportunities for mistakes.

Lists of code tokens are easier to read as actual lists.

I think it was good to have a warning there, it just should be part of the list item.

```suggestion
5. Fake hash: set the hash to one of

   - `""`
   - `lib.fakeHash`
   - `lib.fakeSha256`
   - `lib.fakeSha512`
   
   in the package expression, attempt build and extract correct hash from error messages.

   :::{.warning}
   You must use one of these four fake hashes and not some arbitrarily-chosen hash.
   
   See [](#sec-source-hashes-security).
   :::
```